### ISSUE TITLE
since the last HA core 2024.4.1 update, MCP integration no longer works

### ISSUE THREAD

**Issue by KLM30330** [jpcornil-git/HA-mcp23017#20]
hello, since the last update of HA, I have lost all the entities related to the integration and I cannot get them to work again. when I try to add them I get the following message:
Unable to access mcp23017-0x27 ([Errno 5] I/O error)
Do you have a solution?

**Comment by jpcornil-git** [comment]
Hi,

I just tested with latest/2024.4.3 and everything seems to work as expected.
=> Could you try with 2024.4.3 as well (I guess that you will have the same issue)

What are your hardware (rpi, x86, ...) and software (docker, haos, ...) configurations ?

The error is indicating a connectivity issue with the i2c bus/mcp23017 hardware -> did you verify that nothing changed there (disconnected, mcp23017 address straps, rpi I2c enabled, ...) ?

**Comment by KLM30330** [comment]
Hello, thank you for your response. I reloaded the integration and reload my configuration and everything is back to normal. for information, I am on an RPI 4 with a MiniPiio DIO16 and when I wired this module on the RPI, I had to power the module from the relay board without connecting the RPI power supply. Last day when I tried to restore HA backup,  all m'y relays activated and I had to disconnected the power supply of the relay board so that HA was giving me connexion error... Thanks again for the help, topic resolved!
